cli/internal/ui/ui.go

```diff
@@ -63,13 +57,34 @@ func Rainbow(text string) string {
 	return strings.Join(rainbowStr, "")
 }
 
+type stripAnsiWriter struct {
+	file *os.File
+}
+
+func (into *stripAnsiWriter) Write(src []byte) (n int, err error) {
+	into.file.Write(ansiRegex.ReplaceAll(src, []byte{}))
+	return n, nil
+}
+
 // Default returns the default colored ui
-func Default() *cli.ColoredUi {
+func Default(forceColor ForcedColor) *cli.ColoredUi {
+	applyForcedColor(forceColor);
+	
+	var outWriter, errWriter io.Writer
+
+	if color.NoColor {
+		outWriter = &stripAnsiWriter{file: os.Stdout}
+		errWriter = &stripAnsiWriter{file: os.Stderr}
+	} else {
+		outWriter = os.Stdout
+		errWriter = os.Stderr
+	}
```

Comment:
We are only checking the value of color.NoColor on initialization so changing the value mid run isn't supported. This can be done more dynamically by moving the condition back inside of stripAnsiWriter and always returning that:
```go
func (into *stripAnsiWriter) Write(src []byte) (n int, err error) {
  if color.NoColor {
    src = ansiRegex.ReplaceAll(src, []byte{})
  }
	into.file.Write(src)
	return n, nil
}
```

But this way avoids adding the extra condition every time text gets output. I don't have a strong preference here and I haven't profiled to know what the impact of that would be.